static/app/components/core/compactSelect/compactSelect.tsx

```diff
@@ -154,11 +150,11 @@ export function CompactSelect<Value extends SelectKey>({
           if ('options' in item) {
             return (
               <Section key={item.key} title={item.label}>
-                {item.options.map(opt => (
+                {item.options?.map(opt => (
```

Comment:
I would expect the check above to narrow it down to Array<T>, is this an expected change?

Reply:
I think that's expected as `items` support an explicit `undefined` now?